x-pack/test/api_integration/apis/ml/jobs/datafeed_preview.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const jobId = `fq_datafeed_preview_${Date.now()}`;
+
+  const job = {
+    job_id: `${jobId}_1`,
+    description: '',
+    groups: ['automated', 'farequote'],
+    analysis_config: {
+      bucket_span: '30m',
+      detectors: [{ function: 'distinct_count', field_name: 'airline' }],
+      influencers: [],
+    },
+    data_description: { time_field: '@timestamp' },
+    analysis_limits: { model_memory_limit: '11MB' },
+  };
+
+  function isUpperCase(str: string) {
+    return /^[A-Z]+$/.test(str);
+  }
+
+  function isLowerCase(str: string) {
+    return !/[A-Z]+/.test(str);
+  }
+
+  describe('Datafeed preview', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it(`should return a normal datafeed preview`, async () => {
+      const datafeed = {
+        datafeed_id: '',
+        job_id: '',
+        indices: ['ft_farequote'],
+        query: {
+          bool: {
+            must: [
+              {
+                match_all: {},
+              },
+            ],
+          },
+        },
+        runtime_mappings: {},
+      };
+
+      const { body } = await supertest
+        .post('/api/ml/jobs/datafeed_preview')
```

Comment:
Just curious: Do we hit this endpoint from our UI somewhere? When loading the preview from the job row expansion, I see it's calling a different endpoint: `GET api/ml/datafeeds/datafeed-ec_1/_preview`.

Reply:
Yes it is used when showing the datafeed preview in the job wizards.

![image](https://user-images.githubusercontent.com/22172091/112126362-49035600-8bbc-11eb-8d5b-c7892135fe52.png)